source/geod24/concurrency.d

```diff
@@ -1,967 +1,308 @@
+    /// Ctor
+    public this (string msg = "Owner terminated") @safe pure nothrow @nogc
     {
-        static struct Container { Tid t; }
-        static assert(!hasLocalAliasing!(Tid, Container, int));
-    }
-
-    @safe unittest
-    {
-        /* Issue 20097 */
-        import std.datetime.systime : SysTime;
-        static struct Container { SysTime time; }
-        static assert(!hasLocalAliasing!(SysTime, Container));
+        super(msg);
     }
+}
 
-    enum MsgType
-    {
-        standard,
-        priority,
-        linkDead,
-    }
 
-    struct Message
-    {
-        MsgType type;
-        Variant data;
+/*******************************************************************************
 
-        this(T...)(MsgType t, T vals) if (T.length > 0)
-        {
-            static if (T.length == 1)
-            {
-                type = t;
-                data = vals[0];
-            }
-            else
-            {
-                import std.typecons : Tuple;
+    Encapsulates all implementation-level data needed for scheduling.
 
-                type = t;
-                data = Tuple!(T)(vals);
-            }
-        }
+    When defining a Scheduler, an instance of this struct must be associated
+    with each logical thread.  It contains all implementation-level information
+    needed by the internal API.
 
-        @property auto convertsTo(T...)()
-        {
-            static if (T.length == 1)
-            {
-                return is(T[0] == Variant) || data.convertsTo!(T);
-            }
-            else
-            {
-                import std.typecons : Tuple;
-                return data.convertsTo!(Tuple!(T));
-            }
-        }
+*******************************************************************************/
 
-        @property auto get(T...)()
-        {
-            static if (T.length == 1)
-            {
-                static if (is(T[0] == Variant))
-                    return data;
-                else
-                    return data.get!(T);
-            }
-            else
-            {
-                import std.typecons : Tuple;
-                return data.get!(Tuple!(T));
-            }
-        }
+public struct ThreadInfo
+{
+    /// Transceiver device required for message exchange between threads.
+    public Transceiver     transceiver;
 
-        auto map(Op)(Op op)
-        {
-            alias Args = Parameters!(Op);
+    /// Sheduler
+    public FiberScheduler  scheduler;
 
-            static if (Args.length == 1)
-            {
-                static if (is(Args[0] == Variant))
-                    return op(data);
-                else
-                    return op(data.get!(Args));
-            }
-            else
-            {
-                import std.typecons : Tuple;
-                return op(data.get!(Tuple!(Args)).expand);
-            }
-        }
-    }
+    /// After making the request, wait until the response comes,
+    public WaitingManager  wmanager;
```

Comment:
Why does `concurrency` have knowledge of the `WaitingManager` ?

Reply:
It was removed from here.